samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -0,0 +1,254 @@
+"""
+Custom Lambda Authorizer class definition
+"""
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import List, Optional
+
+from samcli.local.apigw.authorizers.authorizer import Authorizer
+
+
+class IdentitySource(ABC):
+    def __init__(self, identity_source: str):
+        """
+        Abstract class representing an identity source validator
+
+        Paramters
+        ---------
+        identity_source: str
+            The identity source without any prefix
+        """
+        self.identity_source = identity_source
+
+    @abstractmethod
+    def is_valid(self, **kwargs) -> bool:
+        """
+        Validates if the identity source is present
+        """
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, IdentitySource):
+            return NotImplemented
```

Comment:
why not to return false ?

Reply:
I had actually folded this into a single statement on the line just below this one but changed it to what `mypy` suggested to do instead (return NotImplemented). Not sure what I did to trigger that, but looks like if I add that back to the return at the end of this method it is fine now.

Will add this back to the other checks.

Reply:
we could this without ABCs too if it makes it simpler.